api/base/utils.py

```diff
@@ -19,17 +20,24 @@ def absolute_reverse(view_name, query_kwargs=None, args=None, kwargs=None):
     return url
 
 
-def get_object_or_404(model_cls, query_or_pk):
+def get_object_or_error(model_cls, query_or_pk):
+
     if isinstance(query_or_pk, basestring):
         query = Q('_id', 'eq', query_or_pk)
     else:
         query = query_or_pk
     try:
-        return model_cls.find_one(query)
+        obj = model_cls.find_one(query)
+
+        if getattr(obj, 'is_deleted', False) is True :
+            raise Gone
+        if getattr(obj, 'is_active', None) is False:
```

Comment:
i found code in the osf.io repo proving that the default value of 'is_deleted' is False,  i couldnt find anything for 'is_active' but from what i found im pretty sure it is None